### PR TITLE
feat: boilerplate for express server localhost 1337 

### PR DESCRIPTION
**Pull request by linhtran174**

### CODE REVIEW

File: core/src/fs.ts [hunk 1]
```diff
@@ -1,3 +1,9 @@
+const fetchRetry = require("fetch-retry")(global.fetch);
+
+const PORT = 1337;
+const LOCAL_HOST = "127.0.0.1";
+const JAN_HTTP_SERVER_URL = `http://${LOCAL_HOST}:${PORT}`;
+const JAN_FS_API = `${JAN_HTTP_SERVER_URL}/fs`;
```
Comment (louis-jan) [inline core/src/fs.ts:6]: Inject from app, `Core` should not define these.

File: server/package.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
+{
```
Comment (0xSage) [inline server/package.json:1]: Can I ask that you write up a standard README explaining this subdir a bit more?

File: server/v1/index.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import modelsAPI from './models'
+import threadsAPI from './threads'
+
+import { FastifyInstance, FastifyPluginAsync } from 'fastify'
+
+const router: FastifyPluginAsync = async (app: FastifyInstance, opts) => {
+    app.register(
+        assistantsAPI,
+        {
+            prefix: "/assisstants"
```
Comment (0xSage) [inline server/v1/index.ts:12]: Mispelling

**Review by 0xSage — COMMENTED** [review]
Initial code scaffold looks fine. I'm looking for how some of the shared extensions interface & logic (used across both server-only and nativeOS) will be refactored out given this PR. We should aim for as little code duplication as possible.

File: core/src/fs.ts [hunk 4]
```diff
@@ -1,3 +1,5 @@
+const fetchRetry = require("fetch-retry")(global.fetch);
```
[the remark this answers is not in the document]
Comment (linhtran174) [reply core/src/fs.ts:1]: Ty, just done

File: server/v1/index.ts (hunk 3 above)
Comment (linhtran174) [reply server/v1/index.ts:12]: hawkeye! :v, ty, just fixed

File: server/main.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import fastify from 'fastify'
+
+import v1API from './v1'
+const JAN_API_PORT = 1337;
+const server = fastify()
+
+const USER_ROOT_DIR = '.data'
```
[the remark this answers is not in the document]
Comment (linhtran174) [reply server/main.ts:7]: Misspelled, should be './data'. This refers to the `data` folder in the source tree which server is currently used to save data like  - this is temporary hack though, should be configurable but I leave that out for now.

File: server/main.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import fastify from 'fastify'
+
+import v1API from './v1'
+const JAN_API_PORT = 1337;
+const server = fastify()
```
[the remark this answers is not in the document]
Comment (linhtran174) [reply server/main.ts:5]: - Fastify generally have better performance, code size overhead than Express
- Hiro suggested it earlier and Louis found an integration layer which facilitate communication Fastify <-> electron Chrome thread without exposing port, which Express dont seem to have. 
- It's not super important imo but guys prefer it and I'm fine switching to Fastify.

File: server/package.json (hunk 2 above)
Comment (linhtran174) [reply server/package.json:1]: Sure, will add in the next commit

File: server/main.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import fastify from 'fastify'
+
+import v1API from './v1'
+const JAN_API_PORT = 1337;
```
Comment (louis-jan) [inline server/main.ts:4]: move to .env

File: server/package.json [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  "name": "jan-server",
+  "version": "0.1.3",
+  "main": "./build/main.js",
+  "author": "Jan <service@jan.ai>",
+  "license": "AGPL-3.0",
+  "homepage": "https://github.com/janhq/jan/tree/main/electron",
+  "description": "Use offline LLMs with your own data. Run open source models like Llama2 or Falcon on your internal computers/servers.",
+  "build": {
```
Comment (louis-jan) [inline server/package.json:9]: no build action in server package

File: server/package.json [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    "artifactName": "jan-${os}-${arch}-${version}.${ext}"
+  },
+  "scripts": {
+    "lint": "eslint . --ext \".js,.jsx,.ts,.tsx\"",
+    "test:e2e": "playwright test --workers=1",
+    "dev": "nodemon .",
+    "build": "tsc",
+    "build:test": "",
+    "build:publish": ""
```
Comment (louis-jan) [inline server/package.json:57]: remove empty actions

File: server/package.json [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    "@types/request": "^2.48.12",
+    "@uiball/loaders": "^1.3.0",
+    "express": "^4.18.2",
+    "pacote": "^17.0.4",
+    "request": "^2.88.2",
+    "request-progress": "^3.0.0",
+    "use-debounce": "^9.0.4"
+  },
+  "devDependencies": {
+    "@playwright/test": "^1.38.1",
```
Comment (louis-jan) [inline server/package.json:70]: remove electron related stuffs

File: Makefile [hunk 11]
```diff
@@ -15,6 +15,10 @@ endif
 # Installs yarn dependencies and builds core and extensions
 install-and-build: build-uikit
 ifeq ($(OS),Windows_NT)
+	powershell -Command "yarn config set network-timeout 300000; \
+	$$env:NITRO_VERSION = Get-Content .\\plugins\\inference-plugin\\nitro\\version.txt; \
+	Write-Output \"Nitro version: $$env:NITRO_VERSION\"; yarn build:core; yarn install; yarn build:plugins"
```
Comment (louis-jan) [inline Makefile:20]: rebase latest main then do build:extensions instead (terminology update, plugin -> extension)

File: server/main.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+
+
+const JAN_API_PORT = Number.parseInt(process.env.JAN_API_PORT || '1337')
+const JAN_API_HOST = process.env.JAN_API_HOST || "0.0.0.0"
+
+server.listen({
+  port: JAN_API_PORT, 
+  host: JAN_API_HOST
+}).then(() => {
+  console.log(`JAN API listening at: http://0.0.0.0:${JAN_API_PORT}`);
```
Comment (louis-jan) [inline server/main.ts:17]:
```suggestion
  console.log(`JAN API listening at: http://${JAN_API_HOST}:${JAN_API_PORT}`);
```

**Review by louis-jan — APPROVED** [review]
LGTM!

**Review by vuonghoainam — APPROVED** [review]

**Pull request merged**